.pre-commit-config.yaml

```diff
@@ -0,0 +1,51 @@
+# pre-commit is a tool to perform a predefined set of tasks manually and/or
+# automatically before git commits are made.
+#
+# Config reference: https://pre-commit.com/#pre-commit-configyaml---top-level
+#
+# Common tasks
+#
+# - Run on all files:   pre-commit run --all-files
+# - Register git hooks: pre-commit install --install-hooks
+#
+repos:
+  # Autoformat: Python code
+  - repo: https://github.com/ambv/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args: [--target-version=py36]
+
+  # Autoformat: markdown, yaml
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.2.1
+    hooks:
+      - id: prettier
```

Comment:
WDYT about adding nodejs as a dependency to simply *contribute* to pure-Python projects? I love prettier, but I have worried about this. Should we wait for feedback from contributors? The files affected in this repo are pretty minimal.

Reply:
https://pre-commit.com/#node says 
>  node hooks work without any system-level dependencies. It has been tested on linux and macOS and may work under cygwin.

Though I haven't checked how this behaves in practice

Reply:
If we required adding nodejs as a dependency to contribute to a python project, that would be a serious issue in my mind as well.

I have not explicitly tried without having node, but I'm happy to read what @manics quoted.

Reply:
Neat! I just tested it in an empty docker environment, and it uses https://github.com/ekalinin/nodeenv to get node itself. pre-commit set up the prettier environment (including getting node itself!)  quicker than the env for black, so I don't think anyone should notice the runtime. This is wonderful news!